alphadia/workflow/optimization.py

```diff
@@ -95,24 +96,42 @@ def step(
                 verbosity="progress",
             )
             return
+        if record_step:
+            try:
+                # Takes the target number of precursors if at least the target is found,
+                precursor_cutoff_idx = (
+                    precursors_df.sort_values(by="precursor_idx")
+                    .iloc[
+                        2
+                        * self.workflow.config["calibration"][
+                            "optimization_lock_target"
+                        ]
+                    ]
+                    .precursor_idx
+                )
+            except IndexError:
```

Comment:
I know that "favor 'try' over 'upfront check'" is the pythonic way, but I feel in this case it would increase readability if the check was explictly done ..
as far as I got it
```python
num_targets = 2* self.workflow.config["calibration"][
                            "optimization_lock_target"
                        ]
if len(precursors_df) >=  num_targets:
...
else
...
```